[W14-1] BrainTrain

eugenefdw opened the pull request:
@digipie @jeraldtsy @lallanachang @halcyoneee

DigiPie commented:
Looks good! Thanks for the effort @eugenefdw ! 😃

kylase-learning commented:
## Feedback (Week 7; v1.1)

### Github Project Page
- [ ] Update Project description
- [ ] Update Link to Project Page

### README
- [x] UI Mockup updated in project website
- [ ] Make the project website look like a product home page (aka remove the top nav bar)
- [x] AB4 acknowledgement
- [x] Travis badge updated to own fork's
- [x] Coveralls badge updated to own fork's

### [Project Page](https://cs2103-ay1819s2-w14-1.github.io/main/)
- [ ] Update to remove AB4 reference
- [ ] Travis badge updated to own fork's
- [ ] Coveralls badge updated to own fork's

### About Us page
- [x] Updated with photos
- [x] Roles and components assigned

UG/DG feedback will be provided to you in the documents.

kylase-learning commented on build.gradle:
```diff
@@ -156,7 +158,7 @@ test {
 
         if (runGuiTests) {
-            test.include 'systemtests/**'
+        //    test.include 'systemtests/**'
```
I assume that you are commenting this out temporarily?

halcyoneee replied:
Yes, this will be up again when we have an actually working bottom panel GUI

kylase-learning commented on src/main/java/seedu/address/MainApp.java:
```diff
@@ -15,16 +16,18 @@
-import seedu.address.model.util.SampleDataUtil;
-import seedu.address.storage.AddressBookStorage;
-import seedu.address.storage.JsonAddressBookStorage;
+import seedu.address.model.modelmanager.management.ManagementModel;
```
I recalled that some of you mentioned that it is difficult to conform to the ordering imposed by the style checker. 

I think (high probability) that the ordering is in ascending alphabetical order partitioned by each period, i.e. it is sorted by `package` them `subpackage` then `subsubpackage` (in the form of `package.subpackage.subsubpackage`).

There is a rational behind this ordering as if you want to look through the imports it is easier to skim through via this order. 

Hope this helps!

halcyoneee committed:
Merge pull request #173 from DigiPie/v1.4

Card Management System mid-v1.4

jeraldtsy committed:
Merge branch 'v1.4' of https://github.com/cs2103-ay1819s2-w14-1/main into V1.4_User

jeraldtsy committed:
Update parseStringIntoCard function in CsvUserStorage class to become Optional and not throw
Update CsvUserStorageclass test cases

halcyoneee committed:
Merge branch 'v1.4' of https://github.com/cs2103-ay1819s2-w14-1/main into feat/quiz

# Conflicts:
#	src/main/resources/view/DarkTheme.css
#	src/test/java/seedu/address/logic/LogicManagerTest.java
#	src/test/java/seedu/address/model/modelmanager/ManagementModelManagerTest.java

halcyoneee committed:
Update LightTheme according to DarkTheme

jeraldtsy committed:
Update CsvUserStorageTest

jeraldtsy committed:
Update CsvUserStorageTest to have only 1 single folder

jeraldtsy committed:
Add new line to pass travis test

jeraldtsy committed:
Merge pull request #139 from jeraldtsy/V1.4_User

Update CsvUserStorage to sanitise input values

halcyoneee committed:
Remove unused methods in FileUtil, JsonUtil and WebViewUtil

halcyoneee committed:
Update MainWindow and UI handles for system tests

halcyoneee committed:
Add getManagementModelUser in QuizModel

halcyoneee committed:
Skeleton for systemtests

halcyoneee committed:
Add systemtests for Help, ListLesson, OpenLesson and QuizAnswer

halcyoneee committed:
Update build.gradle with systemtests

halcyoneee committed:
Remove BrowserPanelHandle

halcyoneee committed:
Add SampleDataUtil

lallanachang committed:
Solve issue for mid v1.4

halcyoneee committed:
Fix creating file as folder

halcyoneee committed:
Fix 0 lesson size when initLessonList

halcyoneee committed:
Add invalid command system tests

halcyoneee committed:
Merge branch 'v1.4' of https://github.com/cs2103-ay1819s2-w14-1/main into feat/quiz

lallanachang committed:
Update Session for mid-v1.4

lallanachang committed:
Fix style

lallanachang committed:
Update constructors

lallanachang committed:
Solve large count issue

halcyoneee committed:
Replace QuizUiDisplayFormatter with current QuizCard

- QuizCard will hold its own headers and hint

halcyoneee committed:
Add test for SampleDataUtil

halcyoneee committed:
Update QuizAnswerCommand in UGDG

halcyoneee committed:
Update PPP to include other contributions

halcyoneee committed:
Fix codacy style check

halcyoneee committed:
Add more newlines for quiz mode display

halcyoneee committed:
Merge pull request #176 from lallanachang/ui_v1.4

Session mid-v1.4

halcyoneee committed:
Merge branch 'v1.4' of https://github.com/cs2103-ay1819s2-w14-1/main into feat/quiz

# Conflicts:
#	src/main/java/seedu/address/logic/commands/quiz/QuizAnswerCommand.java
#	src/main/java/seedu/address/logic/commands/quiz/QuizStartCommand.java
#	src/main/java/seedu/address/model/quiz/QuizUiDisplayFormatter.java
#	src/main/java/seedu/address/ui/MainPanel.java
#	src/main/resources/view/MainPanel.fxml
#	src/test/java/seedu/address/logic/LogicManagerTest.java
#	src/test/java/seedu/address/logic/commands/quiz/QuizAnswerCommandTest.java
#	src/test/java/seedu/address/model/modelmanager/QuizModelManagerTest.java
#	src/test/java/seedu/address/model/quiz/QuizUiDisplayFormatterTest.java
#	src/test/java/seedu/address/ui/MainPanelTest.java

halcyoneee committed:
Fix total correct answer

halcyoneee committed:
Update to include total correct questions attempted

halcyoneee committed:
Updates MainApp version to v1.3.1

halcyoneee committed:
Fix english

halcyoneee committed:
Merge pull request #175 from halcyoneee/feat/quiz

Quiz mid-v1.4

halcyoneee committed:
Merge pull request #177 from CS2103-AY1819S2-W14-1/v1.4

mid-v1.4

halcyoneee committed:
Add QuizResult UI to show the stats before exiting quiz mode

halcyoneee committed:
Remove unused headers getter and setter

halcyoneee committed:
Add new test for edge case - end with card answered wrongly twice

eugenefdw committed:
Update default data storage locations

Userdata being inside lesson data caused confusion for users. The lesson
data folder has also been renamed to reflect so.

eugenefdw committed:
Update AboutUs to refer to our team instead of se-edu.

DigiPie committed:
Update coverage of management UI

eugenefdw committed:
Add back page header

Updated team name in AboutUs to BrainTrain

[skip ci]

DigiPie committed:
Enforce singleton design for utility classes

which also achieves 100% coverage for utils package

eugenefdw committed:
Update PPP partially

DigiPie committed:
Refactor core utils and tests

DigiPie committed:
Refactor Version and its test

DigiPie committed:
Add GuiSettings JUnit test

DigiPie committed:
Add LogsCenter JUnit test

DigiPie committed:
Refactor VersionTest

halcyoneee committed:
Add legends to footer

halcyoneee committed:
Merge branch 'v1.4' of https://github.com/DigiPie/main into v1.4

DigiPie committed:
Update Lessons Overview and Lesson View

DigiPie committed:
Add SetQuestionAnswerCommand - resolves #118

DigiPie committed:
Merge branch 'v1.4' of https://github.com/DigiPie/main into v1.4

halcyoneee committed:
Refactor QuizAnswerCommand checking

DigiPie committed:
Update GUI

DigiPie committed:
Update GUI

DigiPie committed:
Refactor messages

DigiPie committed:
Update messages

DigiPie committed:
Fix bug in model stub

DigiPie committed:
Update CloseLesson and OpenLesson commands

DigiPie committed:
Various fixes

eugenefdw committed:
Update file format saving for separate header

eugenefdw committed:
Add parsing for header data from file

Remainder of reading cards + fields is not completed

eugenefdw committed:
Update test files to match new format

eugenefdw committed:
Add parsing of field names

DigiPie committed:
Refactor seedu.address.commons.core package

DigiPie committed:
Refactor seedu.address.commons.util package

DigiPie committed:
Update SetTest command

DigiPie committed:
Update Dark and Light themes

DigiPie committed:
Fix CSS bugs

DigiPie committed:
Refactor CSS themes

DigiPie committed:
Refactor ParserUtil to prevent instantiation

eugenefdw committed:
Finish full modification to lesson save data format

DigiPie committed:
Extend StringUtil and update test

eugenefdw committed:
Update tests for new format

DigiPie committed:
Fix label UI component bugs (fixes #182 fixes #181)

eugenefdw committed:
Update all missing optional handles for readLessonList

DigiPie committed:
Update UI tests

DigiPie committed:
Improve contrast of light theme

DigiPie committed:
Update AddCardCommandTest

DigiPie committed:
Add SetTestCommand-related tests

DigiPie committed:
Refactor `seedu.address.commons.core` and `seedu.address.commons.util` packages

eugenefdw committed:
Update UG

DigiPie committed:
Update ManagementModeParserTest and ManagmentModelManagerTest

DigiPie committed:
Update tests

DigiPie committed:
Refactor OpenLessonCommand command word

From `openLesson` to `open`

DigiPie committed:
Refactor CloseLessonCommand command word

From `closeLesson` to `close`

DigiPie committed:
Refactor Open/Close messages

DigiPie committed:
Update AddLessonParserTest

DigiPie committed:
Update LessonOverviewTest

DigiPie committed:
Update LessonCardTest

DigiPie committed:
Update Syntax

jeraldtsy committed:
Update jeraldtsy.adoc

eugenefdw committed:
Update PPP

eugenefdw committed:
Update PPP, UG

DigiPie committed:
Add additional scripts for local testing

DigiPie committed:
Refactor `openLesson` command to `edit` command

DigiPie committed:
Refactor `closeLesson` command to `save` command

DigiPie committed:
Update messages

DigiPie committed:
Refactor `reloadLessons` to `reload`

jeraldtsy committed:
Update UserGuide for User transfer file feature

DigiPie committed:
Refactor core c/ and optional o/ parameters

Changed to test value t/ and hint value h/ parameters

DigiPie committed:
Update themes

jeraldtsy committed:
Update UGDG with userstorage explanation

DigiPie committed:
Implements result box colour change to indicate failed command

DigiPie committed:
Refactor `addLesson` command to use `\t` and `\h` params

DigiPie committed:
Refactor `setTest` command word to `set`

DigiPie committed:
Remove unused parameters `q/` and `a/`

DigiPie committed:
Refactor `save` command word to `quit` command word

halcyoneee committed:
Fix extension style not showing

DigiPie committed:
Add UI fixes

DigiPie committed:
Fix system tests

DigiPie committed:
Remove unstable tests

DigiPie committed:
Fix AddLessonParserTest

DigiPie committed:
Update messages

lallanachang committed:
Update for v1.4

eugenefdw committed:
Update DG

DigiPie committed:
Refactor `SampleDataUtil`, adds `SampleLessons` and `SampleCards`

DigiPie committed:
Fixes #184 - now able to delete sample lesson

DigiPie committed:
Add another sample lesson

halcyoneee committed:
Updating UGDG for quiz mode

lallanachang committed:
Update start command and ui

DigiPie committed:
Update AboutUs

DigiPie committed:
Update Personal Portfolio Page

DigiPie committed:
Update messages

lallanachang committed:
Update comments

halcyoneee committed:
Update use case and manual testing in DG

jeraldtsy committed:
Update documents to fit style

halcyoneee committed:
Update UGDG for changeTheme

DigiPie committed:
Update EditLessonCommandSystemTest

DigiPie committed:
Fix EditLessonCommandSystemTest

DigiPie committed:
Force Codacy test

eugenefdw committed:
Update DG

lallanachang committed:
Fix errors

lallanachang committed:
Fix style error

DigiPie committed:
Merge branch 'ui_v1.4' of https://github.com/lallanachang/main into lallanachang_ui_v1.4

DigiPie committed:
Resolve merge conflict

DigiPie committed:
Fix bugs

DigiPie committed:
Fix #187 bug

DigiPie committed:
Replace screenshot with bug inside with screenshot with no bug

DigiPie committed:
Update digipie.adoc

halcyoneee committed:
Merge pull request #183 from jeraldtsy/V1.4_User

Update documentation

lallanachang committed:
Update UG

lallanachang committed:
Update UG in PPP

lallanachang committed:
Fix bugs #188

halcyoneee committed:
Quiz code clean up

DigiPie committed:
Update UG/DG

DigiPie committed:
Fix critical lesson/card parsing bugs

eugenefdw committed:
Merge branch 'v1.4' of https://github.com/CS2103-AY1819S2-W14-1/main into v1.4

# Conflicts:
#	docs/UserGuide.adoc

DigiPie committed:
Resolve merge conflict

DigiPie committed:
Update docs/DeveloperGuide.adoc

Co-Authored-By: eugenefdw <31066291+eugenefdw@users.noreply.github.com>

DigiPie committed:
Update docs/DeveloperGuide.adoc

Co-Authored-By: eugenefdw <31066291+eugenefdw@users.noreply.github.com>

DigiPie committed:
Update docs/DeveloperGuide.adoc

Co-Authored-By: eugenefdw <31066291+eugenefdw@users.noreply.github.com>

DigiPie committed:
Update docs/DeveloperGuide.adoc

Co-Authored-By: eugenefdw <31066291+eugenefdw@users.noreply.github.com>

DigiPie committed:
Merge pull request #179 from eugenefdw/v1.4

Storage and Documentation v1.4

DigiPie committed:
Resolve merge conflicts

halcyoneee committed:
Rename to QuizAnswerIntegrationTest

halcyoneee committed:
Add QuizAnswerCommand with stub

halcyoneee committed:
Add description for quiz commands

halcyoneee committed:
Format quiz javadoc

halcyoneee committed:
Merge branch 'v1.4' into ui_v1.4

halcyoneee committed:
Merge pull request #185 from lallanachang/ui_v1.4

Session v1.4

halcyoneee committed:
Update ChangeTheme test to use stub

halcyoneee committed:
Merge branch 'v1.4' of https://github.com/cs2103-ay1819s2-w14-1/main into feat/quiz

# Conflicts:
#	docs/UserGuide.adoc

lallanachang committed:
Update PPP

lallanachang committed:
Merge branch 'v1.4' of https://github.com/CS2103-AY1819S2-W14-1/main into ui_v1.4

eugenefdw committed:
Update DG

DigiPie committed:
Resolve merge conflict

DigiPie committed:
Resolve merge conflict

DigiPie committed:
Update QuizStartCommand

jeraldtsy committed:
Update DG

halcyoneee committed:
Merge pull request #180 from DigiPie/v1.4

Card Management System v1.4

halcyoneee committed:
Merge branch 'v1.4' of https://github.com/cs2103-ay1819s2-w14-1/main into feat/quiz

# Conflicts:
#	src/test/java/systemtests/QuizAnswerCommandSystemTest.java

eugenefdw committed:
Update docs

halcyoneee committed:
Fix expectMessage for wrongModel

halcyoneee committed:
Refactor QuizStartCommand to StartCommand

halcyoneee committed:
Rename executeActual to executeQuiz

eugenefdw committed:
Update PPP

halcyoneee committed:
Remove Suggested programming task from DG

lallanachang committed:
Update DG and PPP

lallanachang committed:
Merge branch 'v1.4' of https://github.com/CS2103-AY1819S2-W14-1/main into ui_v1.4

halcyoneee committed:
Update font to non light

halcyoneee committed:
Remove unused BrowserPanel

halcyoneee committed:
Update DG component diagrams

lallanachang committed:
Fix style

halcyoneee committed:
Update DG UG PPP for quiz mode

lallanachang committed:
Fix style

halcyoneee committed:
Add EOF at PPP

DigiPie committed:
Merge pull request #186 from halcyoneee/feat/quiz

Quiz v1.4

DigiPie committed:
Update images

lallanachang committed:
Update

lallanachang committed:
Merge branch 'v1.4' of https://github.com/CS2103-AY1819S2-W14-1/main into ui_v1.4

halcyoneee committed:
Update quiz tag and section details

DigiPie committed:
Update PPP/UG

lallanachang committed:
Update graph

lallanachang committed:
Update DG

lallanachang committed:
Update PPP

lallanachang committed:
Update graph

lallanachang committed:
Fix style

eugenefdw committed:
Update docs

eugenefdw committed:
Merge branch 'v1.4' of https://github.com/CS2103-AY1819S2-W14-1/main into v1.4

lallanachang committed:
Update PPP

lallanachang committed:
Fix style

jeraldtsy committed:
Merge branch 'v1.4' of https://github.com/cs2103-ay1819s2-w14-1/main into V1.4_User

# Conflicts:
#	docs/DeveloperGuide.adoc

halcyoneee committed:
Merge branch 'v1.4' of https://github.com/cs2103-ay1819s2-w14-1/main into feat/quiz

halcyoneee committed:
Update UGDG and about us

jeraldtsy committed:
Update AboutUs roles
Update DG by adding code snippet
Update jeraldtsy.adoc
Update CSVUserStorage.java to remove magic numbers

jeraldtsy committed:
Update jeraldtsy.adoc

lallanachang committed:
Merge branch 'v1.4' of https://github.com/CS2103-AY1819S2-W14-1/main into ui_v1.4

jeraldtsy committed:
Update DG

lallanachang committed:
Merge branch 'v1.4' of https://github.com/CS2103-AY1819S2-W14-1/main into ui_v1.4

DigiPie committed:
Update UG/DG

eugenefdw committed:
Update UG and PPP

DigiPie committed:
Update UG/DG

eugenefdw committed:
Merge branch 'v1.4' of https://github.com/CS2103-AY1819S2-W14-1/main into v1.4

# Conflicts:
#	docs/UserGuide.adoc

lallanachang committed:
Improve graph quality

halcyoneee committed:
Update PPP

eugenefdw committed:
Merge branch 'v1.4' of https://github.com/CS2103-AY1819S2-W14-1/main into v1.4

halcyoneee committed:
Merge branch 'v1.4' of https://github.com/cs2103-ay1819s2-w14-1/main into feat/quiz

eugenefdw committed:
Merge branch 'v1.4' of https://github.com/CS2103-AY1819S2-W14-1/main into v1.4

jeraldtsy committed:
Update DG

jeraldtsy committed:
Merge branch 'v1.4' of https://github.com/cs2103-ay1819s2-w14-1/main into V1.4_User

jeraldtsy committed:
Update DG

halcyoneee committed:
Update DG sentence case and glossary

halcyoneee committed:
Merge branch 'v1.4' of https://github.com/cs2103-ay1819s2-w14-1/main into feat/quiz

# Conflicts:
#	docs/DeveloperGuide.adoc

DigiPie committed:
Update UG/DG

DigiPie committed:
Resolve merge conflicts

eugenefdw committed:
Transparent background for UserStorageSequenceDiagram

DigiPie committed:
Update UG/DG

DigiPie committed:
Merge branch 'v1.4' of https://github.com/cs2103-ay1819s2-w14-1/main into v1.4

DigiPie committed:
Update UG/DG

DigiPie committed:
Update Messages strings

To include `start` and `changeTheme` commands

eugenefdw committed:
Update UG grammar

halcyoneee committed:
Update user stories and sequence diagrams

eugenefdw committed:
Add intro to DG

eugenefdw committed:
Merge branch 'v1.4' of https://github.com/CS2103-AY1819S2-W14-1/main into v1.4

halcyoneee committed:
Update command header

DigiPie committed:
Update UG/DG

DigiPie committed:
Merge branch 'v1.4' of https://github.com/cs2103-ay1819s2-w14-1/main into v1.4

DigiPie committed:
Update DG

DigiPie committed:
Update UG/DG

DigiPie committed:
Update to UG/DG

DigiPie committed:
Final update to UG/DG

DigiPie committed:
Merge pull request #190 from CS2103-AY1819S2-W14-1/v1.4

v1.4
